mesonbuild/compilers/c.py

```diff
@@ -710,3 +710,79 @@ def get_include_args(self, path: str, is_system: bool) -> T.List[str]:
         if path == '':
             path = '.'
         return ['--include_path=' + path]
+
+class TCCCCompiler(CCompiler):
+    LINKER_PREFIX = '-Wl,'
+
+    def __init__(self, exelist: T.List[str], version: str, for_machine: MachineChoice,
+                 is_cross: bool, info: 'MachineInfo',
+                 exe_wrapper: T.Optional['ExternalProgram'] = None,
+                 linker: T.Optional['DynamicLinker'] = None,
+                 full_version: T.Optional[str] = None):
+        self.id = 'tcc'
+        CCompiler.__init__(self, exelist, version, for_machine, is_cross,
+                           info, exe_wrapper, linker=linker, full_version=full_version)
+        self.can_compile_suffixes.add('s')
+        self.base_options = {
+            OptionKey(o) for o in ['b_ndebug', 'b_staticpic']}
+
+    def get_options(self) -> 'KeyedOptionDictType':
+        opts = CCompiler.get_options(self)
+        key = OptionKey('std', machine=self.for_machine, lang=self.language)
+        opts[key].choices = ['none', 'c99', 'c11']
+        return opts
+
+    def get_always_args(self) -> T.List[str]:
+        return []
+
+    def get_option_compile_args(self, options: 'KeyedOptionDictType') -> T.List[str]:
+        args = []
+        std = options[OptionKey('std', lang=self.language, machine=self.for_machine)]
+        if std.value != 'none':
+            args.append('-std=' + std.value)
+        return args
+
+    def get_dependency_gen_args(self, outtarget: str, outfile: str) -> T.List[str]:
+        return ['-MD', '-MF', outfile]
+
+    def get_optimization_args(self, optimization_level: str) -> T.List[str]:
+        optimization_args = {
+            '0': [],
+            'g': ['-O0', '-g'],
+            '1': ['-O1'],
+            '2': ['-O2'],
+            '3': ['-O3'],
+            's': ['-Os'],
+        }  # type: T.Dict[str, T.List[str]]
+        return optimization_args[optimization_level]
```

Comment:
> Note: GCC options -Ox, -fx and -mx are ignored.

https://bellard.org/tcc/tcc-doc.html#Option-summary

Looking at the tinycc source, the optimizations flag is parsed (as one would expect according to `tcc -hh`), but it doesn't seem to be used at all (as one would infer from the documentation).